### PR TITLE
feat: add teacher dashboard

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link

[Teacher Dashboard page](https://www.notion.so/uwblueprintexecs/Teacher-Dashboard-page-3d48a79f5d9342f5afc7f6d56fe230d5)

## Implementation description
image

- Fix popover component to better match designs
- Add generic popover for simple cases
- Add teacher dashboard header
- Add dashboard sections to page

## Steps to test
1. Sign in as a teacher and view your homepage.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/common/info/messages/EmptyClassroomsGoToPage.tsx [hunk 1]
```diff
@@ -10,7 +10,8 @@ import MessageContainer from "../MessageContainer";
 const EmptyClassroomsGoToPageMessage = (): React.ReactElement => {
   const styles = useStyleConfig("Center", { variant: "emptyMessage" });
   const history = useHistory();
-  const onClick = () => history.push(Routes.CLASSROOMS_PAGE);
+  const onClick = () =>
+    history.push(Routes.CLASSROOMS_PAGE, { openAddClassroomModal: true });
```
Comment (carissa-tang) [inline frontend/src/components/common/info/messages/EmptyClassroomsGoToPage.tsx]:
```suggestion
    history.push(Routes.CLASSROOMS_PAGE, { isAddClassroomModalOpen: true });
```
nit renaming

File: frontend/src/components/pages/teacher/TeacherDashboardPage.tsx [hunk 2]
```diff
@@ -1,8 +1,78 @@
...
+            </Text>
+            <Spacer />
+            <RouterLink
+              color="blue.300"
+              fontWeight="bold"
+              textDecor="none"
+              to={Routes.DISPLAY_ASSESSMENTS_PAGE}
+            >
+              View All
+            </RouterLink>
```
Comment (carissa-tang) [inline frontend/src/components/pages/teacher/TeacherDashboardPage.tsx]: map to avoid repetition?

**Review by carissa-tang — COMMENTED** [review]
some small comments

File: frontend/src/components/pages/teacher/TeacherDashboardPage.tsx (hunk 2 above)
Comment (jfdoming) [reply frontend/src/components/pages/teacher/TeacherDashboardPage.tsx:69]: made a custom component instead due to slight variations in the two sections

**Review by carissa-tang — APPROVED** [review]
lgtm!

**Pull request merged**